From 2463eda385df8bc1bb3fea798b32ab1dd7757438 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 8 Jan 2021 13:37:25 +0000
Subject: [PATCH 348/634] arch/arm: Add __memset alias to memset_rpi.S

memset_rpi.S is an optimised memset implementation, but doesn't define
__memset (which was just added to memset.S). As a result, building
for the BCM2835 platform causes a link failure.

Add __memset as yet another alias to our common implementation.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/lib/memset_rpi.S | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/lib/memset_rpi.S b/arch/arm/lib/memset_rpi.S
index e8469cecabc1..bda4c5b99367 100644
--- a/arch/arm/lib/memset_rpi.S
+++ b/arch/arm/lib/memset_rpi.S
@@ -52,6 +52,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 ENTRY(mmioset)
 ENTRY(memset)
+ENTRY(__memset)
 ENTRY(__memset32)
 ENTRY(__memset64)
 
@@ -124,5 +125,6 @@ ENTRY(__memset64)
         .unreq  DAT3
 ENDPROC(__memset64)
 ENDPROC(__memset32)
+ENDPROC(__memset)
 ENDPROC(memset)
 ENDPROC(mmioset)
-- 
2.33.1

